### PR TITLE
Various refactors and improvements

### LINKED ISSUE #182: Controllers spawned through ros2 not working for Webots: in Windows 10
I installed Webots and ros2 on my Win10 system. I tried to install the webots_ros2 pkg on my local workspace and launched the demo file `ros2 launch webots_ros2_universal_robot universal_robot_multiple.launch.py` for UR robot simulation. Webots opens and the conveyor belt starts moving but the extern controllers for ur robots do not work and hence the robots don't move. 

When I press CTRL+C in PowerShell I get the following output

```
Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

Try the new cross-platform PowerShell https://aka.ms/pscore6

Loading personal and system profiles took 1135ms.
PS C:\Users\Admin> ros2 launch webots_ros2_universal_robot universal_robot_multiple.launch.py
[INFO] [launch]: All log files can be found below C:\Users\Admin\.ros/log\2021-01-02-19-28-48-433426-DESKTOP-Q4VN964-13284
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [universal_robot_multiple.wbt --batch --mode=realtime-1]: process started with pid [7848]
[INFO] [webots_robotic_arm_node.EXE-2]: process started with pid [2640]
[INFO] [webots_robotic_arm_node.EXE-3]: process started with pid [8512]
[universal_robot_multiple.wbt --batch --mode=realtime-1] OpenGL error: GL_INVALID_ENUM
[universal_robot_multiple.wbt --batch --mode=realtime-1] OpenGL error: GL_INVALID_ENUM
[universal_robot_multiple.wbt --batch --mode=realtime-1] OpenGL error: GL_INVALID_ENUM
[WARNING] [launch]: user interrupted with ctrl-c (SIGINT)
[ERROR] [universal_robot_multiple.wbt --batch --mode=realtime-1]: process has died [pid 7848, exit code 3221225786, cmd 'C:\Users\Admin\AppData\Local\Programs\Webots\msys64\mingw64\bin\webots C:\Users\Admin\Documents\ros2_ws\install\share\webots_ros2_universal_robot\worlds\universal_robot_multiple.wbt --batch --mode=realtime'].
[universal_robot_multiple.wbt --batch --mode=realtime-1] ^C
[ERROR] [webots_robotic_arm_node.EXE-3]: process has died [pid 8512, exit code 1, cmd 'C:\Users\Admin\Documents\ros2_ws\install\lib\webots_ros2_core\webots_robotic_arm_node.EXE --webots-robot-name=UR5e --ros-args -r __ns:=/UR5e --params-file C:\Users\Admin\AppData\Local\Temp\launch_params_tzxm7hn3'].
[webots_robotic_arm_node.EXE-3] **Cannot open file: C:\Users\Admin\AppData\Local\Temp/webots-13688/WEBOTS_SERVER (retrying in 1 second)**

[ERROR] [webots_robotic_arm_node.EXE-2]: process has died [pid 2640, exit code 1, cmd 'C:\Users\Admin\Documents\ros2_ws\install\lib\webots_ros2_core\webots_robotic_arm_node.EXE --webots-robot-name=UR3e --ros-args -r __ns:=/UR3e --params-file C:\Users\Admin\AppData\Local\Temp\launch_params_8wummeqt'].
[webots_robotic_arm_node.EXE-2] **Cannot open file: C:\Users\Admin\AppData\Local\Temp/webots-13688/WEBOTS_SERVER (retrying in 1 second)**

PS C:\Users\Admin>
```

I am unable to understand why this **Cannot open file: C:\Users\Admin\AppData\Local\Temp/webots-13688/WEBOTS_SERVER (retrying in 1 second)** message shows up for each controller spawn up! I tried running from Powershell Admin but got similar output. 

For once (maybe by chance) it started working and I was able to see the robots moving but then subsequent trials were not working.

PS: I had to modify `webots_ros2_core/webots_ros2_core/utils.py` file in order to get the launch file working in the first place. `webotsw.exe` executable path was not added to Path system variable so I had to do that manually first. Then made the following changes to run the webots application through ros2 launch.

1.  Added `from pathlib import Path` and replaced `os.environ['HOME']` in line 68 with `str(Path.home())`. `os.environ['HOME']` was giving a KeyError in python3.
2.  Replaced `subprocess.check_output(['which', 'webots'])` in line 59 with `subprocess.check_output(['where', 'webots'])` because `which` is available in Linux. Windows alternative was `where` command. 

If the above changes might have been the cause for the issue I am facing please let me know. Any help will be highly appreciated :). Thanks 

**My System config**
 - Webots Version:  R2021a
 - ROS Version: Foxy
 - Operating System: Windows 10 (Both Webots and ROS2 are installed in Windows 10)
 - Graphics Card: AMD Radon M360 (pretty old one but it works)

### LINKED ISSUE #185: Webots_ros2 requires specific Webots version
**Describe the Bug**
I found, that webots_ros2 only works with the exact version of Webots specified here:
https://github.com/cyberbotics/webots_ros2/blob/master/webots_ros2_core/webots_ros2_core/utils.py#L35

As of yesterday, the master branch of webots_ros2 required 2020b,
I built the develop branch, so I could use 2021a, However this did not work, as I had compiled the latest develop branch of Webots "2021b"
So I recompiled the master branch of Webots. It still did not work, as webots_ros2 requires "2021a" and I now had "2021 revision 1"

**Expected behavior**
- building both master branches together should work
- webots_ros2 should accept newer versions (perhaps with a warning)

**Proposed solutions**
- Turn the hard rquirement into a soft requirement, allowing the user to use any version they want and simply warn them if not the expected version was found. The popup window should probably include the Version that was actually found
- Add multiple valid versions to cover develop and built versions. This could look like this:

```python
def get_required_webots_version():
    """Return the Webots version compatible with this version of the package."""
    return ['R2021a', 'R2021a revision 1', 'R2021b']
```

### PR DESCRIPTION
**Pull request by lukicdarkoo**
This PR has to do the following:
- [x] Make the package work on Windows (fixes #182) 
  - [x] Get rid of `tkinter`, so the package can be used comfortably with [ROS on Windows](https://github.com/ms-iot/ROSOnWindows).
  - [x] Add progress bar for downloading 
  - [x] Fix the home path on Windows 
  - [x] Replace `which` with `where` on Windows 
- [x] Introduce the notion of minimum and target Webots version (fixes #185).
- [x] Deprecate `TfPublisher` in favor of our new transform publisher based on `get_urdf()`. 
- [x] Enforce the camel-case project-wide (and other PEP conventions) and restructure the project a bit.
- [x] Add camera acceleration
- [x] Add Rolling CI
- [x] Verify regressions ([here](https://github.com/cyberbotics/webots_ros2/wiki/Internal-Tests))
- [x] Verify auto-download
- [x] Add to changelog

### CODE REVIEW

**Comment by lukicdarkoo** [comment]
This is ready for review as soon as it is confirmed to solve #188.

If we merge this today it will be available in the next Foxy sync:
https://discourse.ros.org/t/preparing-for-foxy-sync-2020-01-07/18268

File: webots_ros2_core/webots_ros2_core/utils.py [hunk 1]
```diff
@@ -19,95 +19,172 @@
...
+    except FileNotFoundError:
+        # The `which` command not available
+        pass
+
+    # Look at standard installation pathes
+    paths = [
+        '/usr/local/webots',                                    # Linux default install
+        '/snap/webots/current/usr/share/webots',                # Linux snap install
+        '/Applications/Webots.app',                             # macOS default install
+        r'C:\\Program Files\\Webots',                           # Windows default install
```
Comment (omichel) [inline webots_ros2_core/webots_ros2_core/utils.py:161]: That looks wrong, it should be either:
```suggestion
        'C:\\Program Files\\Webots',                           # Windows default install
```
or
```suggestion
        r'C:\Program Files\Webots',                           # Windows default install
```

**Review by omichel — CHANGES_REQUESTED** [review]

File: webots_ros2_core/webots_ros2_core/utils.py (hunk 1 above)
Comment (lukicdarkoo) [reply webots_ros2_core/webots_ros2_core/utils.py:161]: It worked in tests, but it makes sense not to combine those two

**Review by omichel — APPROVED** [review]
Maybe we can wait for the feedback of @anastasiapan to merge this PR?

**Pull request merged**